src/components/v2/Pagination/index.stories.tsx

```diff
@@ -0,0 +1,68 @@
+/** @jsxImportSource @emotion/react */
+import React from 'react';
+import { ComponentMeta, Story } from '@storybook/react';
+import { withThemeProvider, withCenterStory } from 'stories/decorators';
+import Typography from '@mui/material/Typography';
+import Box from '@mui/material/Box';
+import { PALETTE } from 'theme/MuiThemeProvider/muiTheme';
+
+import { Table } from '../Table';
+import { ITableRowProps } from '../Table/types';
+import { columns, rows, useTableStyles } from '../Table/storiesUtils';
+
+import { usePagination } from './usePagination';
+import { Pagination } from '.';
+
+export default {
+  title: 'Components/Pagination',
+  component: Pagination,
+  decorators: [withThemeProvider, withCenterStory({ width: 800 })],
+  parameters: {
+    backgrounds: {
+      default: PALETTE.background.default,
+    },
+  },
+} as ComponentMeta<typeof Pagination>;
+
+const PaginationWithCustomRowsPerPageTemplate: Story<{
+  initialPageNumber: number;
+  rowsPerPageCount: number;
+}> = ({ initialPageNumber, rowsPerPageCount }) => {
+  const styles = useTableStyles();
+  const data = [...rows, ...rows, ...rows];
+  const { pagesCount, currentPageIndex, setCurrentPageIndex, currentPageData, itemsCountString } =
+    usePagination<ITableRowProps[]>({
+      data,
+      initialPageNumber,
+      rowsPerPageCount,
+    });
+
+  return (
+    <>
+      <Table
+        columns={columns}
+        data={currentPageData}
+        title="Market Data"
+        minWidth="650px"
+        rowKeyIndex={0}
+        tableCss={styles.tableCss}
+        cardsCss={styles.cardsCss}
+        css={styles.table}
+      />
```

Comment:
Just to avoid confusion, let's only include the `Pagination` component in the story.

Reply:
but that's the way to see how it works with data because the hook usePagination is actually part of the component

Reply:
I can leave some comments in code for clarifying if this would solve the issue

Reply:
`Pagination` doesn't need to be coupled to the table component. In the future we might want to use pagination without rendering a table, and so the component should be more modular than that.
Pagination is also handled by the backend: its goal is so that we don't have to load all of the data, but we can load it page by page. In your current implementation, the full data would be loaded then divided into pages which will present performance issues.

Perhaps the specs needed to be more explicit, I didn't think it would cause issues.
I'll update the ticket then.

Reply:
I updated the ticket with more specs. Let me know if it's still unclear

Reply:
done